### PR TITLE
User signup and login front side

### PR DESCRIPTION
**Pull request by tmznwnel00**
- User can signup with email, password, nickname(optional), gender(optional), location(optional)
- Check email function with regular expression
- User can login with email and password (lastLoginAt field is not updating)
- Email verify

** server needs optional fields also (should be fixed)

### CODE REVIEW

File: client/src/pages/Signup.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        전라남도: ["목포시", "여수시", "순천시", "나주시", "광양시", "담양군", "곡성군", "구례군", "고흥군", "보성군", "화순군", "장흥군", "강진군", "해남군", "영암군", "무안군", "함평군", "영광군", "장성군", "완도군", "진도군", "신안군"],
+        경상북도: ["포항시", "경주시", "김천시", "안동시", "구미시", "영주시", "영천시", "상주시", "문경시", "경산시", "군위군", "의성군", "청송군", "영양군", "영덕군", "청도군", "고령군", "성주군", "칠곡군", "예천군", "봉화군", "울진군", "울릉군"],
+        경상남도: ["창원시", "마산시", "진주시", "진해시", "통영시", "사천시", "김해시", "밀양시", "거제시", "양산시", "의령군", "함안군", "창녕군", "고성군", "남해군", "하동군", "산청군", "함양군", "거창군", "합천군"],
+        제주특별자치도: ["제주시", "서귀포시", "북제주군", "남제주군"]
+    }
+
+    const emailCheck = (email) => {
+        return emailRegEx.test(email);
+    }
+    const onChange1 = (e) => {
```
Comment (sungminoh) [inline client/src/pages/Signup.js]: onChangeEmail

**Review by sungminoh — APPROVED** [review]

File: server/config.py [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+from dotenv import load_dotenv
+
+load_dotenv()
 
 JWT_SECRET_KEY = os.getenv('JWT_SECRET_KEY')
 JWT_ALGORITHM = "HS256"
 JWT_ACCESS_TOKEN_EXPIRE_MINUTES = 30
+
+EMAIL_ADDRESS = os.getenv('EMAIL_ADDRESS')
+EMAIL_PASSWORD = os.getenv('EMAIL_APP_PASSWORD')
```
Comment (sungminoh) [inline server/config.py:12]: what is this for?

Comment (tmznwnel00) [reply server/config.py:12]: @sungminoh for STMP(Single Mail Transfer Protocol), it need sender email credentials.
I newly created gmail for this.

File: client/src/pages/Signup.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    const [password, setPassword] = useState('');
+    const [nickname, setNickname] = useState('');
+    const [gender, setGender] = useState('');
+    const [location, setLocation] = useState('시/도');
+    const [subLocation, setSubLocation] = useState('');
+    const [checkbox1, setCheckbox1] = useState(false);
+    const [checkbox2, setCheckbox2] = useState(false);
+    const navigate = useNavigate();
+
+    const koreaLocation = {
```
Comment (tmznwnel00) [inline client/src/pages/Signup.js]: @sungminoh Doesn't need to relocate koreaLocation to config file?

File: client/src/pages/Signup.js (hunk 1 above)
Comment (tmznwnel00) [reply client/src/pages/Signup.js]: [6d983e4](https://github.com/tmznwnel00/date_dive/pull/6/commits/6d983e459f1ba71485c0659f71481646069c0509)

File: client/src/pages/Signup.js (hunk 3 above)
Comment (tmznwnel00) [reply client/src/pages/Signup.js]: [c3b52e1](https://github.com/tmznwnel00/date_dive/pull/6/commits/c3b52e1a318941544f1e93fcc8286c0d31609dc3)

**Pull request merged**